OpenRA.Mods.Common/Activities/LayMines.cs

```diff
@@ -82,9 +90,12 @@ public override bool Tick(Actor self)
 					return false;
 				}
 
-				LayMine(self);
-				QueueChild(new Wait(20)); // A little wait after placing each mine, for show
-				minefield.Remove(self.Location);
+				if (!StartLayingMine(self))
+				{
+					layingMine = true;
+					QueueChild(new Wait(minelayer.Info.LayMineTime, false));
+				}
```

Comment:
This also means we wait the `LayMaineTime` when there is no ammo pool available.

With the suggestion from my previous comment this can become:
```suggestion
				if (!StartLayingMine(self))
					return false;

				if (minelayer.Info.LayMineTime == 0)
					FinishLayingMine(self);
				else
				{
					layingMine = true;
					QueueChild(new Wait(minelayer.Info.LayMineTime, false));
				}
```

Reply:
or instead `FinishLayingMine` could become a bool just like `StartLayingMine` and them the queueing up of Wait activity could be pushed back into `Tick`

Reply:
Implemented tweaks. I did keep the check for `LayMineTime` in `Tick`, because it might not be obvious, why `FinishLayingMine` couldn't finish laying the mine. With the `LayMineTime` check and queuing up `Wait` close to each other this looks very clear (at least to me).

Also it's not the same case as `StartLayingMine`, because `Tick` doesn't do anything (just returns false), when `StartLayingMine` returns false.